### PR TITLE
Implement shl and shr_s for v128

### PR DESCRIPTION
**Pull request by ngzhian**
shr_u is a bit more tricky due for I8 and I16. E.g. -128 in I8 has all
top bits set (sign extended) in an int32, and shr_u using Int's
implementation will result in 0xffffffc0, instead of 0x000000c0. But we
can't simply change the implementation, since parsing relies on this
behavior. I'll figure that out in a later patch.

### CODE REVIEW

File: interpreter/syntax/operators.ml [hunk 1]
```diff
@@ -227,6 +227,8 @@ let v128_bitselect = Ternary (V128Op.Bitselect)
 let i8x16_neg = Unary (V128 (V128Op.I8x16 V128Op.Neg))
 let i8x16_any_true = Test (V128 (V128Op.I8x16 V128Op.AnyTrue))
 let i8x16_all_true = Test (V128 (V128Op.I8x16 V128Op.AllTrue))
+let i8x16_shl = Shift (V128Op.I8x16 V128Op.Shl)
+let i8x16_shr_s = Shift (V128Op.I8x16 V128Op.ShrS)
```
Comment (rossberg) [inline interpreter/syntax/operators.ml:231]: Nit: could shorten this to `Shift V128Op.(I8x16 Shl)`. Similarly many of the other cases.

File: interpreter/syntax/ast.ml [hunk 2]
```diff
@@ -135,6 +138,7 @@ and instr' =
   | Ternary of ternop                 (* ternary numeric operator *)
   | Convert of cvtop                  (* conversion *)
   | ExtractLane of extractop          (* extract lane from v128 value *)
+  | Shift of shiftop                  (* shifts for v128 value *)
```
Comment (rossberg) [inline interpreter/syntax/ast.ml:141]: Perhaps rename this to SimdShift for clarity? And symmetrically, ExtractLane -> SimdLane?

File: interpreter/syntax/operators.ml (hunk 1 above)
Comment (ngzhian) [reply interpreter/syntax/operators.ml:231]: Good tip, I'll change only this for now, then make the change for the rest (to reduce some local merge conflicts).

File: interpreter/syntax/ast.ml (hunk 2 above)
Comment (ngzhian) [reply interpreter/syntax/ast.ml:141]: I'll rename Shift here, and extractlane can go in another PR i have for implementing more of extract lane, thanks!

**Comment by ngzhian** [comment]
Force push due to rebebasing.
